salt/tests/unit/modules/files/test_metalk8s_service_configuration.yaml

```diff
@@ -0,0 +1,135 @@
+get_service_config:
+  #. Success: a retrieved Configmap object with `config.yaml` in the data section is merged with a `default_csc` correctly
+  - configmap_name: 'my_configmap'
+    default_csc:
+      apiVersion: my_apiVersion1
+      kind: my_kind1
+      spec:
+        replicas: my_replicas
+    configmap_obj:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: my_configmap
+      data:
+        config.yaml: |-
+          kind: my_kind2
+          apiVersion: my_apiVersion2
+    result:
+      apiVersion: my_apiVersion2
+      kind: my_kind2
+      spec:
+        replicas: my_replicas
+
+  #. Success: a retrieved Configmap object with `config.yaml` in the data
+  #  section is merged with a `default_csc` correctly given the apiVersion and
+  #  kind from args matches the corresponding value in `config.yaml` section
+  - configmap_name: 'my_configmap'
+    apiVersion: my_apiVersion4
+    kind: my_kind4
+    default_csc:
+      apiVersion: my_apiVersion3
+      kind: my_kind3
+      spec:
+        replicas: my_replicas
+    configmap_obj:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: my_configmap
+      data:
+        config.yaml: |-
+          kind: my_kind4
+          apiVersion: my_apiVersion4
```

Comment:
To me kind and apiVersion is exactly the kind of value you do not want to update, add several other keys and update one of them please